src/map.cpp

```diff
@@ -4741,6 +4741,7 @@ std::list<item> map::use_amount( const tripoint &origin, const int range, const
 template <typename Stack>
 std::list<item> use_charges_from_stack( Stack stack, const itype_id type, long &quantity)
 {
+    // TODO: Fix the bug with this function consuming liquid off the ground
     std::list<item> ret;
     for( auto a = stack.begin(); a != stack.end() && quantity > 0; ) {
         if( a->use_charges(type, quantity, ret) ) {
```

Comment:
> Do the inherent liquids that don't show up in 'e', like water pump's or water tile's, count for your check?

This check is only for items on the square. Furniture/terrain that is able to produce water does not contain water items, so it won't apply here.